src/pykeen/metrics/ranking.py

```diff
@@ -356,13 +565,67 @@ def expected_value(self, num_candidates: np.ndarray, num_samples: Optional[int]
             the expected Hits@k value
         """
         return (
-            self.k
-            * np.mean(np.reciprocal(np.asanyarray(num_candidates, dtype=float)).clip(min=None, max=1 / self.k)).item()
-        )
+            self.k * np.reciprocal(np.asanyarray(num_candidates, dtype=float)).clip(min=None, max=1.0 / self.k).mean()
+        ).item()
+
+    def variance(
+        self,
+        num_candidates: np.ndarray,
+        num_samples: Optional[int] = None,
+    ) -> float:
+        e = self.expected_value(num_candidates=num_candidates, num_samples=num_samples)
+        return e * (1 - e)
 
 
 @parse_docdata
-class AdjustedArithmeticMeanRank(ArithmeticMeanRank):
+class AdjustedHitsAtK(HitsAtK):
+    """The adjusted Hits at K ($AH_k$).
+
+    ---
+    link: https://github.com/pykeen/pykeen/pull/814
+    description: The re-indexed adjusted hits at K
+    """
+
+    name = "Adjusted Hits at K"
+    # Maybe need to calculate this based on num_candidates
+    value_range = ValueRange(lower=None, lower_inclusive=False, upper=1, upper_inclusive=True)
+    synonyms = (
+        "ahk",
+        "ah@k",
+        "ahits@k",
+        "ah@",
+        "ahits@",
+        "ahits_at_",
+        "ah_at_",
+        "adjusted_hits_at_",
+    )
+    increasing = True
+    supported_rank_types = (RANK_REALISTIC,)
+    needs_candidates = True
+
+    def __call__(self, ranks: np.ndarray, num_candidates: Optional[np.ndarray] = None) -> float:  # noqa: D102
+        ev = super().expected_value(num_candidates=num_candidates)
+        return (super().__call__(ranks=ranks, num_candidates=num_candidates) - ev) / max(1 - ev, EPSILON)
+
+
+@parse_docdata
+class ZHitsAtK(DecreasingZMixin, HitsAtK):
+    """The z-scored hits at k ($ZAH_k$).
+
+    ---
+    link: https://github.com/pykeen/pykeen/pull/814
+    description: The z-scored hits at K
+    """
+
+    name = "z-Hits at K"
+    synonyms = ("z_hits_at_", "zahk")
+    increasing = True
+    supported_rank_types = (RANK_REALISTIC,)
+    needs_candidates = True
```

Comment:
does it make sense to move those to `DecreasingZMixin`?

Reply:
I made a base class in a6577585